pkg/automationconfig/automation_config_builder.go

```diff
@@ -124,7 +130,12 @@ func (b *Builder) Build() (AutomationConfig, error) {
 
 		process := newProcess(toHostName(b.name, i), h, b.mongodbVersion, b.name, opts...)
 		processes[i] = process
-		members[i] = newReplicaSetMember(process, i)
+
+		if b.replicaSetHorizons != nil {
+			members[i] = newReplicaSetMember(process, i, b.replicaSetHorizons[i])
+		} else {
+			members[i] = newReplicaSetMember(process, i, nil)
```

Comment:
I believe this is line that is causing the test failures, when a null 'horizons' value is passed to the mongodb agents, it puts them into a bad state. I think the simplest fix for this is to update the field declaration to 
```golang
Horizons    ReplicaSetHorizons `json:"horizons,omitempty"
```

to prevent the null serialization.

Reply:
Applied the change and tested e2e locally. Thanks! :D